### PR TITLE
Builtin: Add a new built-in functin MAKEDATE

### PR DESCRIPTION
**Pull request by dawxy**
Add a new built-in functin MAKEDATE
#236

### CODE REVIEW

**Comment by ngaut** [comment]
@zhexuany @bobotu Please feel free to review more PRs.

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1740,7 +1740,38 @@ type builtinMakeDateSig struct {
...
 func (b *builtinMakeDateSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("MAKEDATE")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, nil
+	}
+	year := args[0].GetInt64()
```
Comment (shenli) [inline expression/builtin_time.go]: Use args[0].ToInt64(). MySQL also accept other type of input:
mysql> select makedate("71", 1);
+-------------------+
| makedate("71", 1) |
+-------------------+
| 1971-01-01        |
+-------------------+
1 row in set (0.00 sec)

File: expression/builtin_time.go [hunk 2]
```diff
@@ -1740,7 +1740,38 @@ type builtinMakeDateSig struct {
...
+	dayOfYear := args[1].GetInt64()
```
Comment (shenli) [inline expression/builtin_time.go]: So as here.

**Comment by dawxy** [comment]
@shenli Well, has been repaired, thank you

**Comment by shenli** [comment]
LGTM
@dawxy Thanks for your PR!

**Pull request merged**